app-modules/ai/src/Filament/Resources/AiAssistantResource/Pages/EditAiAssistant.php

```diff
@@ -106,6 +106,17 @@ public function getSaveFormAction(): Action
             });
     }
 
+    public static function canAccess(array $parameters = []): bool
+    {
+        $assistant = $parameters['record'];
+
+        if ($assistant->is_default == true) {
```

Comment:
Let's add the following tests for this:

- If a user has the correct permissions, make sure we get a forbidden response if the assistant is default
- If a user has the correct permissions, make sure we get a successful response if the assistant is not default